server/dal/src/storage_engine.rs

```diff
@@ -109,29 +99,36 @@ impl Storage {
     pub async fn get_photo_url_by_id<S: AsRef<str>>(
         &self,
         photo_id: S,
-        photo_quality: PhotoQuality,
+        photo_quality: &PhotoQuality,
     ) -> Result<String, error::StorageError> {
-        let response = self
-            .client
-            .get_object()
-            .bucket(&self.bucket_name)
-            .key(Self::format_id_with_quality(
-                photo_id.as_ref(),
-                photo_quality,
-            ))
-            .presigned(PresigningConfig::expires_in(Duration::from_secs(6000))?)
-            .await?;
-
-        let url = String::from(response.uri().to_string().split('?').next().unwrap());
+        let url = format!(
+            "https://{}.s3.amazonaws.com/{}_{}",
```

Comment:
Hard coding the URL breaks other S3 service than amazon, most notably MinIO. Furthermore, the S3_FORCE_PATH_STYLE variable is completely ignored with this implementation, ultimately breaking any local deployment using the MinIO container (which is used by default)